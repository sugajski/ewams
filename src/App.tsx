import React from 'react';
import {NativeRouter, Switch, Route} from 'react-router-native';
import {Provider} from 'mobx-react';
import {Provider as PaperProvider} from 'react-native-paper';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import {Dashboard} from './routes';
import {WeatherStore} from './stores';

const weatherStore = new WeatherStore();

export default () => {
  return (
    <SafeAreaProvider>
      <Provider weatherStore={weatherStore}>
        <PaperProvider>
          <NativeRouter>
            <Switch>
              <Route exact path="/" component={Dashboard} />
            </Switch>
          </NativeRouter>
        </PaperProvider>
      </Provider>
    </SafeAreaProvider>
  );
};
