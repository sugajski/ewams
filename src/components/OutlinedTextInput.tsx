import React, {FC, forwardRef} from 'react';
import {TextInputProps} from 'react-native';
import {TextInput} from 'react-native-paper';

import {Colors} from '../themes';
import OutlinedTextInputStyles from './styles/OutlinedTextInputStyles';

interface IOutlinedTextInputProps extends TextInputProps {
  label: string;
  error: boolean;
  ref: any;
}

const textInputTheme = {
  colors: {
    primary: Colors.sapphire,
    accent: Colors.sapphire,
    error: Colors.magenta,
    text: Colors.black,
    background: Colors.white,
    placeholder: Colors.slate,
  },
};

const OutlinedTextInput: FC<IOutlinedTextInputProps> = forwardRef(
  (props, ref) => (
    //@ts-ignore warning caused by react-native-paper types being incompatible with newest react-native types
    <TextInput
      mode="outlined"
      style={OutlinedTextInputStyles.input}
      theme={textInputTheme}
      //@ts-ignore warning spreading props will override ref, but it's not
      ref={ref}
      {...props}
    />
  ),
);

export default OutlinedTextInput;
