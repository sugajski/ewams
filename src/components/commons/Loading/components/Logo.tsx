import React, {FC} from 'react';
import {View, ViewProps} from 'react-native';

import LogoStyles from './styles/LogoStyles';

const Logo: FC<ViewProps> = () => (
  <View style={LogoStyles.container}>
    <View style={[LogoStyles.square, LogoStyles.a]} />
    <View style={[LogoStyles.square, LogoStyles.b]} />
    <View style={[LogoStyles.square, LogoStyles.c]} />
    <View style={[LogoStyles.square, LogoStyles.d]} />
  </View>
);

export default Logo;
