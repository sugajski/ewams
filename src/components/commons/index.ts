import DismissKeyboardWrapper from './DismissKeyboardWrapper';
import Loading from './Loading/Loading';
import Separator from './Separator';

export {DismissKeyboardWrapper, Loading, Separator};
