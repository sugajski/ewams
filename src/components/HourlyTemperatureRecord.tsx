import React, {FC} from 'react';
import {View, Text} from 'react-native';

import {Separator} from './commons';
import {Localizable} from '../lib';
import HourlyTemperatureRecordStyles from './styles/HourlyTemperatureRecordStyles';

interface IRecord {
  hour: string;
  temperature: string;
}

interface IHourlyTemperatureRecordProps {
  record: IRecord;
  index: number;
}

const HourlyTemperatureRecord: FC<IHourlyTemperatureRecordProps> = ({
  record,
  index,
}) => (
  <View style={HourlyTemperatureRecordStyles.container}>
    <Text style={HourlyTemperatureRecordStyles.text}>
      {index === 0 ? Localizable.t('weatherDetails.now') : record.hour}
    </Text>
    <Separator />
    <Text style={HourlyTemperatureRecordStyles.text}>{record.temperature}</Text>
  </View>
);

export default HourlyTemperatureRecord;
