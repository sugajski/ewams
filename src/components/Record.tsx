import React, {FC} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import moment from 'moment';

import {Colors} from '../themes';
import {Separator} from './commons';
import RecordStyles from './styles/RecordStyles';

interface IRecord {
  city: string;
  date: moment.Moment;
}

interface IRecordProps {
  record: IRecord;
  checkCurrentForecast: (city: string) => void;
  isLast: boolean;
}

const Record: FC<IRecordProps> = ({record, checkCurrentForecast, isLast}) => (
  <TouchableOpacity onPress={() => checkCurrentForecast(record.city)}>
    <View style={RecordStyles.container}>
      <Text style={RecordStyles.title}>{record.city}</Text>
      <Text style={RecordStyles.date}>{moment(record.date).fromNow()}</Text>
    </View>
    {!isLast && <Separator color={Colors.slate} />}
  </TouchableOpacity>
);

export default Record;
