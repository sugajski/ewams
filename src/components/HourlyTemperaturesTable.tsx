import React, {FC} from 'react';
import {ScrollView} from 'react-native';

import HourlyTemperatureRecord from './HourlyTemperatureRecord';
import HourlyTemperaturesTableStyles from './styles/HourlyTemperaturesTableStyles';

interface IHourlyTemperaturesRecord {
  hour: string;
  temperature: string;
}

interface IHourlyTemperaturesTableProps {
  hourlyTemperatures: IHourlyTemperaturesRecord[];
}

const HourlyTemperaturesTable: FC<IHourlyTemperaturesTableProps> = ({
  hourlyTemperatures,
}) => (
  <ScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    contentContainerStyle={HourlyTemperaturesTableStyles.container}>
    {hourlyTemperatures.map((record, index) => (
      <HourlyTemperatureRecord key={index} record={record} index={index} />
    ))}
  </ScrollView>
);

export default HourlyTemperaturesTable;
