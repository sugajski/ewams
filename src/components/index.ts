import OutlinedTextInput from './OutlinedTextInput';
import Record from './Record';
import SendButton from './SendButton';
import SearchHistory from './SearchHistory';

export {OutlinedTextInput, Record, SendButton, SearchHistory};
