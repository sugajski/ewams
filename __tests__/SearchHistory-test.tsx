import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';

import {SearchHistory} from '../src/components';

const searchHistoryPropsMock = {
  history: [
    {city: 'Warszawa', date: moment()},
    {city: 'KrakÃ³w', date: moment('2011-06-05')},
  ],
  checkCurrentForecast: () => console.log('record clicked!'),
};

test('SearchHistory renders correctly', () => {
  const tree = renderer
    .create(
      <SearchHistory
        history={searchHistoryPropsMock.history}
        checkCurrentForecast={searchHistoryPropsMock.checkCurrentForecast}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
