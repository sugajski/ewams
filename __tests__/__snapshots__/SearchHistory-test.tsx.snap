// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHistory renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontFamily": "OpenSans",
        "fontSize": 24,
        "lineHeight": 40,
      }
    }
  >
    Ostatnio wyszukiwane:
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#444b59",
            "fontFamily": "OpenSans",
            "fontSize": 18,
            "lineHeight": 24,
          }
        }
      >
        Warszawa
      </Text>
      <Text
        style={
          Object {
            "color": "#858896",
            "fontFamily": "OpenSans",
            "fontSize": 12,
            "lineHeight": 16,
          }
        }
      >
        kilka sekund temu
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 1,
            "textShadowColor": "#00000066",
            "textShadowOffset": Object {
              "height": 1,
              "width": -1,
            },
            "textShadowRadius": 6,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#9a9eaf",
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#444b59",
            "fontFamily": "OpenSans",
            "fontSize": 18,
            "lineHeight": 24,
          }
        }
      >
        Krak√≥w
      </Text>
      <Text
        style={
          Object {
            "color": "#858896",
            "fontFamily": "OpenSans",
            "fontSize": 12,
            "lineHeight": 16,
          }
        }
      >
        9 lat temu
      </Text>
    </View>
  </View>
</View>
`;
